evap/staff/importers/json.py

```diff
@@ -346,15 +382,27 @@ def _import_evaluation(self, course: Course, data: ImportEvent) -> Evaluation:
         else:
             course_end = max(datetime.strptime(app["end"], self.DATETIME_FORMAT) for app in data["appointments"])
 
+        name_de, name_en = "", ""
+        try:
+            evaluation = Evaluation.objects.get(course=course, cms_id=data["gguid"])
+        except Evaluation.DoesNotExist:
+            evaluation = None
+
         if data["isexam"]:
             # Set evaluation time frame of three days for exam evaluations:
             evaluation_start_datetime = course_end.replace(hour=8, minute=0, second=0, microsecond=0) + timedelta(
                 days=1
             )
             evaluation_end_date = (course_end + timedelta(days=3)).date()
 
-            name_de = data["title"].split(" - ")[-1] if " - " in data["title"] else "Prüfung"
-            name_en = data["title_en"].split(" - ")[-1] if " - " in data["title_en"] else "Exam"
+            exam_type = self._get_exam_type(data["type"])
+            if not evaluation:
+                name_de = self._disambiguate_name(
+                    exam_type.name_de, list(course.evaluations.values_list("name_de", flat=True))
+                )
+                name_en = self._disambiguate_name(
+                    exam_type.name_en, list(course.evaluations.values_list("name_en", flat=True))
+                )
```

Comment:
This degrades to O(n^2) operations. Maybe fine for now, I think we have superlinear runtime in other parts here as well?